Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -27,18 +28,21 @@ public final class DiagnosticConsoleWriter: DiagnosticFormattingConsumer {
     ///   - formattingOptions: The formatting options for the diagnostics.
     ///   - baseUrl: A url to be used as a base url when formatting diagnostic source path.
     ///   - highlight: Whether or not to highlight the default diagnostic formatting output.
+    ///   - fileManager: The file manager that reads the contents of source files.
     public init(
         _ stream: TextOutputStream = LogHandle.standardError,
         formattingOptions options: DiagnosticFormattingOptions = [],
         baseURL: URL? = nil,
-        highlight: Bool? = nil
+        highlight: Bool? = nil,
+        fileManager: FileManagerProtocol = FileManager.default
```

Comment:
If we don't want to expose _Common as public API then we need to figure out another strategy for this initializer.

Reply:
Essentially if we don't want to expose _Common in our API surface then we need to make sure that it isn't used in any public API declarations. I am not sure but I think that FileManagerProtocol is potentially a useful thing to expose though, so maybe another strategy is required.

Reply:
Hm... it may be best to discuss what direction we want to go in with the team. 

I did avoid exposing _Common in https://github.com/apple/swift-docc/pull/756/commits/4fc6a1644e06eab3fd723b63eda9fb7dac5d1dcf so that we can change `FileManagerProtocol` from `public` to `package` in the future without breaking any public API but we would need to be careful so that we don't introduce new breaking points accidentally. 

As an alternative we could bump the Swift version to 5.9 now and make `FileManagerProtocol` package accessible. We can always increase the visibility in the future bug going is a lot more time and work.